docs/source/reconciliation.rst

```diff
@@ -0,0 +1,134 @@
+Reconciliation
+==============
+Reconciliation is the high-level process of modifying an (extended) prefix map with
+domain-specific rules. This is important as it allows for building on existing
+(extended) prefix maps without having to start from scratch. Further, storing the
+rules to transform an existing prefix map allows for high-level discussion about
+the differences and their reasons.
+
+As a specific example, the `Bioregistry <https://bioregistry.io/>`_ uses ``snomedct`` as a preferred prefix for
+the Systematized Nomenclature of Medicine - Clinical Terms (SNOMED-CT). The
+OBO Foundry community prefers to use ``SCTID`` as the preferred prefix for this
+resource. Rather than maintaining a different extended prefix map than the Bioregistry,
+the OBO Foundry community could enumerate its preferred modifications to the base
+(extended) prefix map, then create its prefix map by transforming the Bioregistry's.
+
+Similarly, a consumer of the OBO Foundry prefix map who's implementing a resolver might want to override the URI prefix
+associated with the `Ontology of Vaccine Adverse Events (OVAE) <https://bioregistry.io/registry/ovae>`_
+to point towards the Ontology Lookup Service instead of the default OntoBee.
+
+There are two operations that are useful for transforming an existing (extended) prefix
+map:
+
+1. **Remapping** is when a given CURIE or URI prefix is replaced with another. See :func:`curies.remap`.
+2. **Rewiring** is when the correspondence between a CURIE prefix and URI prefix is updated. See :func:`curies.rewire`.
+
+Throughout this document, we're going to use the following extended prefix map as an example
+to illustrate how these operations work from a high level.
+
+.. code-block:: json
+
+    [
+        {"prefix": "a", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a1"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+
+CURIE Prefix Remapping
+----------------------
+CURIE prefix remapping is configured by a mapping from existing CURIE prefixes to new CURIE prefixes.
+The following rules are applied for each pair of old/new prefixes:
+
+1. New prefix exists
+~~~~~~~~~~~~~~~~~~~~
+If the new prefix appears as a prefix synonym in the record corresponding to the old prefix, they are swapped.
+This means applying the CURIE prefix remapping ``{"a": "a1"}`` results in the following
+
+.. code-block:: json
+
+    [
+        {"prefix": "a1", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a"]},
+        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
+    ]
+
+If the new prefix appears as a preferred prefix or prefix synonym for any other record, one of two things can happen:
+
+1. Do nothing (lenient)
+2. Raise an exception (strict)
+
+This means applying the CURIE prefix remapping ``{"a": "b"}`` results in either no change or an exception being raised.
+
+2. New prefix doesn't exist, old prefix is a preferred prefix
```

Comment:
## _Another remapping case:_ New prefix doesn't exist, old prefix is a prefix synonym
e.g. Remapping `{"a1": "c"}` could result in:
```json
    [
        {"prefix": "c", "uri_prefix": "https://example.org/a/", "prefix_synonyms": ["a", "a1"]},
        {"prefix": "b", "uri_prefix": "https://example.org/b/"}
    ]
```

In this case example, `c` is the new prefix, and `a1` is the prefix synonym.

What do you think? Should this be the behavior?

Reply:
yes, in fact that example is directly below where you commented. I changed the title to be more inclusive

Reply:
Oh darn, dk how I missed that!